daemon/docker_watcher.go

```diff
@@ -249,6 +255,21 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 			}
 		}
 
+		if ep != nil {
+			ep.Mutex.Lock()
+			// Docker appends '/' to container names.
+			ep.DockerContainerName = strings.Trim(containerName, "/")
+			if d.conf.IsK8sEnabled() {
+				if dockerContainer.Config != nil {
+					podNamespace := k8sDockerLbls.GetPodNamespace(dockerContainer.Config.Labels)
+					podName := k8sDockerLbls.GetPodName(dockerContainer.Config.Labels)
```

Comment:
These calls to `GetPodNamespace` and `GetPodName` are lookups into the map `dockerContainer.Config.Labels` for `io.kubernetes.pod.name` and `io.kubernetes.pod.namespace` - so they are not calls to the API server.